pkg/rules/handlers.go

```diff
@@ -0,0 +1,267 @@
+package rules
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/http/types"
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/go-chi/chi/v5"
+)
+
+type API struct {
+	service ServiceInterface
+
+	logger logging.LoggerInterface
+}
+
+func (a *API) RegisterEndpoints(mux *chi.Mux) {
+	mux.Get("/api/v0/rules", a.handleList)
+	mux.Get("/api/v0/rules/{id}", a.handleDetail)
+	mux.Post("/api/v0/rules/{id}", a.handleCreate)
+	mux.Put("/api/v0/rules/{id}", a.handleUpdate)
+	mux.Delete("/api/v0/rules/{id}", a.handleRemove)
+}
+
+func (a *API) handleList(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	limit, err := strconv.ParseInt(r.URL.Query().Get("limit"), 10, 64)
```

Comment:
look for the pagination in here https://github.com/canonical/identity-platform-admin-ui/blob/main/internal/http/types/generic.go

Reply:
Done. The query to admin-ui uses page and size parameters, but the outgoing request to oathkeeper uses limit and offset.